### PR TITLE
Convert some adapters to Kotlin

### PR DESCRIPTION
**Pull request by charlag**
During timeilne refactoring I realized how annoying existing Java code is and that we should get rid of it.
I tried to not make conflicts with timeline branch so there's a bunch of other adapters.
Protip: if you look at diff from second commit only it will actually show you the proper diff.

This is ofc in no rush and can wait.

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/adapter/ThreadAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            VIEW_TYPE_STATUS_DETAILED -> {
+                val view = LayoutInflater.from(parent.context)
+                    .inflate(R.layout.item_status_detailed, parent, false)
+                StatusDetailedViewHolder(view)
+            }
+            else -> {
+                val view = LayoutInflater.from(parent.context)
+                    .inflate(R.layout.item_status, parent, false)
+                StatusViewHolder(view)
+            }
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/ThreadAdapter.kt:49]: I'd either put this in a if/else to not have 3 branches or throw an exception in the third branch (also in the other adapters)

File: app/src/main/java/com/keylesspalace/tusky/adapter/ThreadAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        return if (position == detailedStatusPosition) {
+            VIEW_TYPE_STATUS_DETAILED
+        } else {
+            VIEW_TYPE_STATUS
+        }
+    }
+
+    override fun getItemCount(): Int {
+        return statuses.size
+    }
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/ThreadAdapter.kt:68]:
```suggestion
    override fun getItemCount() = statuses.size
```

File: app/src/main/java/com/keylesspalace/tusky/adapter/AccountAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,92 @@
+/* Copyright 2017 Andrew Dawson
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/AccountAdapter.kt:1]: Maybe update these to "2021 Tusky Contributors"?

**Review by connyduck — COMMENTED** [review]
Looks pretty good!

File: app/src/main/java/com/keylesspalace/tusky/adapter/AccountAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    override fun getItemViewType(position: Int): Int {
+        return if (position == accountList.size && bottomLoading) {
+            VIEW_TYPE_FOOTER
+        } else {
+            VIEW_TYPE_ACCOUNT
+        }
+    }
+
+    fun update(newAccounts: List<Account>) {
+        accountList = removeDuplicates(newAccounts)
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/AccountAdapter.kt:44]: Kotlin has a built in function for this https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/distinct.html

Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/adapter/AccountAdapter.kt:44]: It produces immutable list and we need a mutable one and if I re-implement it or add toMutableList() it won't be simpler

**Review by connyduck — APPROVED** [review]

**Pull request merged**